zebra-state/src/service/chain_tip.rs

```diff
@@ -212,7 +191,11 @@ impl ChainTipSender {
     /// An update is only sent if the current best tip is different from the last best tip
     /// that was sent.
     fn update(&mut self, new_tip: Option<ChainTipBlock>) {
-        let needs_update = match (new_tip.as_ref(), self.active_value.as_ref()) {
+        // Correctness: the `self.sender.borrow()` must not be placed in a `let` binding to prevent
+        // a read-lock being created and living beyond the `self.sender.send(..)` call. If that
+        // happens, the `send` method will attempt to obtain a write-lock and will dead-lock.
+        // Without the binding, the guard is dropped at the end of the expression.
+        let needs_update = match (new_tip.as_ref(), self.sender.borrow().as_ref()) {
```

Comment:
Blocker: We need to test this change for locking bugs.

I started a full sync test on Google Cloud using:
https://github.com/ZcashFoundation/zebra/actions/workflows/test-full-sync.yml

It is running at:
https://github.com/ZcashFoundation/zebra/actions/runs/2222831862